### PR TITLE
[CS2113-T11-2] IHospital

### PR DESCRIPTION
**Pull request by Demonshaha**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -8,22 +8,72 @@
...
+### Exit program
+Step 1: User type "bye" as input to exit the program. IHospital will call Parser#getCommand()
+to return the user command received.
+
+Step 2: After checking the user command is "bye", IHospital will
+call Ui#sayGoodbye to print Goodbye message
+
+The following sequence diagram shows how the exit operation works:
+
+![](ByeCommand.png)
```
Comment (froststein) [inline docs/DeveloperGuide.md:25]: image
For the sequence diagram, consider naming it as fullCommand:datatype, assuming if your return from parser is a string, you could put fullCommand:String instead. 

image
Similarly for your last 2 return arrows containing the Goodbye message you could consider putting it as "message:String" instead.

image
For the arguments in the methods for saveDoctorList(), savePatientList(), saveApppointmentList() consider putting the data type as well. For example, "doctors:ArrayList<Doctors>" if your data type is `ArrayList<Doctors>`.

image
All of the object calls are missing the activation bar, consider adding them.

image
Consider removing the extra boxes that appear at the bottom, I think PlantUML can remove it, if not so you may have to remove it manually.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -8,22 +8,72 @@
...
+Step 2:
+The parser will call the database storing patient information to update the change.
+
+The flow is presented in the sequence diagram (will be included later after pull request is permitted):
+
+
+### Exit program
+Step 1: User type "bye" as input to exit the program. IHospital will call Parser#getCommand()
+to return the user command received.
+
```
Comment (froststein) [inline docs/DeveloperGuide.md:19]: Consider explaining that upon exit, the data is saved, so that all the steps in the sequence diagram is explained.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -8,22 +8,72 @@
...
+### Delete Doctor
+Step 1: User types "delete" to delete a specific doctor from the system
+This will be passed into Duke. Duke will pass the comment to the parser,
+which will parse through the command to remove the doctor.
+
+Step 2: The parser will call on deleteDoctor to the doctor list.
+The doctor list acts as a database storage containing the information of a list of doctors.
+Once the doctor is removed from the doctor list, the system informs the user that the doctor is successfully removed
+
+![](DeleteDoctor.jpg)
```
Comment (froststein) [inline docs/DeveloperGuide.md]: Consider using the same software as the previous 2 diagrams, I understand that NTU provides visual paradigm, but it would be good to standardize the diagrams across the DG.

From what I know the numbering is auto added when using visual paradigm, so consider removing it? If I am not wrong, its not wrong to have numbering, but you may have to check with Prof if the numberings are allowed.

image
The alignment for 1.1.1:deleteDoctor() is a little off, consider moving it higher.

image
The return arrows from methods should be dotted lines instead of solid lines.

You may want to check with Prof if there is a need for an activation bar for the User.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -8,22 +8,72 @@
...
+Parser#parse to understand that User wants to add a doctor, and Parser#parse will return an 
+AddDoctorCommand object.
+
+Step 2: IHospital will call execute of the AddDoctorCommand object and AddDoctorCommand will
+create a new Doctor with the given information and add it into DoctorList using 
+DoctorList#addDoctor(doctor).
+
+The following sequence diagram shows how the add operation works:
+
+![](AddDoctor.png)
```
Comment (froststein) [inline docs/DeveloperGuide.md:39]: image
Similarly for here, the activation bars are missing.
image
Similar mistakes as the above diagram, where the return arrow should have the datatype.

image
Similarly, should remove the bottom boxes as well.

Perhaps you could have a line break for long comments so that you diagram would not appear very long. As the current diagram is a little difficult to read.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -8,22 +8,72 @@
...
+Step 2: After checking the user command is "bye", IHospital will
+call Ui#sayGoodbye to print Goodbye message
+
+The following sequence diagram shows how the exit operation works:
+
+![](ByeCommand.png)
+
+### Add Doctor
+Step 1: User type add command to add a doctor into the system and provide information required.
+IHospital will call Parser#getCommand() to return the user command received, it will then call
```
Comment (froststein) [inline docs/DeveloperGuide.md:29]: Consider renaming the Parser#getCommand() to `Parser#getCommand`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -8,22 +8,72 @@
...
+
+Step 2: After checking the user command is "bye", IHospital will
+call Ui#sayGoodbye to print Goodbye message
+
+The following sequence diagram shows how the exit operation works:
+
+![](ByeCommand.png)
+
+### Add Doctor
+Step 1: User type add command to add a doctor into the system and provide information required.
```
Comment (froststein) [inline docs/DeveloperGuide.md:28]: Perhaps add an example of the command instead of writing it in plain English.

E.g. User types in `add doctor` to add a doctor....

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -8,22 +8,72 @@
...
+IHospital will call Parser#getCommand() to return the user command received, it will then call
+Parser#parse to understand that User wants to add a doctor, and Parser#parse will return an 
+AddDoctorCommand object.
+
+Step 2: IHospital will call execute of the AddDoctorCommand object and AddDoctorCommand will
+create a new Doctor with the given information and add it into DoctorList using 
+DoctorList#addDoctor(doctor).
```
Comment (froststein) [inline docs/DeveloperGuide.md:35]: Similarly for here, DoctorList#addDoctor(doctor) could be `DoctorList#addDoctor`

**Review by froststein — COMMENTED** [review]
Generally this is a good start for a developer guide, just need to clean up on the sequence diagrams, and add more explanations for each command.

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -8,22 +8,72 @@
...
+### Delete Doctor
+Step 1: User types "delete" to delete a specific doctor from the system
+This will be passed into Duke. Duke will pass the comment to the parser,
+which will parse through the command to remove the doctor.
+
+Step 2: The parser will call on deleteDoctor to the doctor list.
+The doctor list acts as a database storage containing the information of a list of doctors.
+Once the doctor is removed from the doctor list, the system informs the user that the doctor is successfully removed
+
+![](DeleteDoctor.jpg)
```
Comment (chihyingho) [inline docs/DeveloperGuide.md]: Perhaps you could consider using PlantUML consistently for all diagrams throughout the DG?

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -8,22 +8,72 @@
...
+The parser will call the database storing patient information to update the change.
+
+The flow is presented in the sequence diagram (will be included later after pull request is permitted):
+![](patient.png)
+
+### Exit program
+Step 1: User type "bye" as input to exit the program. IHospital will call Parser#getCommand()
+to return the user command received.
+
+Step 2: After checking the user command is "bye", IHospital will
```
Comment (chihyingho) [inline docs/DeveloperGuide.md:20]: Perhaps you could explain the save methods that are run before exit?

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -8,22 +8,72 @@
...
+Parser#parse to understand that User wants to add a doctor, and Parser#parse will return an 
+AddDoctorCommand object.
+
+Step 2: IHospital will call execute of the AddDoctorCommand object and AddDoctorCommand will
+create a new Doctor with the given information and add it into DoctorList using 
+DoctorList#addDoctor(doctor).
+
+The following sequence diagram shows how the add operation works:
+
+![](AddDoctor.png)
```
Comment (chihyingho) [inline docs/DeveloperGuide.md:39]: The text in this PlantUML diagram is slightly too small, perhaps you could consider reformatting the diagram such that the text size is consistent with the DG's body font size?
